src/overlays/actors/ovl_En_Encount2/z_en_encount2.c

```diff
@@ -21,26 +28,214 @@ const ActorInit En_Encount2_InitVars = {
     (ActorFunc)EnEncount2_Update,
     (ActorFunc)EnEncount2_Draw
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Encount2_0x808E1560/EnEncount2_Init.asm")
+static ColliderJntSphElementInit sJntSphElementsInit[1] = { // D_808E1EF0
+    {
+        { ELEMTYPE_UNK0, { 0xF7CFFFFF, 0x00, 0x00 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_ON, },
+        { 1, { { 0, 0, 0 }, 0 }, 1 },
+    },
+};
+
+static ColliderJntSphInit sJntSphInit = { // D_808E1F14 
+    { COLTYPE_HARD, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_1, COLSHAPE_JNTSPH, },
+    1, sJntSphElementsInit,
+};
+
+DamageTable damageTable[] = {
+    0xF0, 0xF0, 0x00, 0xF0, 0xE1, 0xE1, 0x00, 0xE1, 0xF0, 0xF0, 0xF0, 0xE1, 0xE1, 0xE1, 0xF0, 0xF0, 
+    0xE1, 0xF0, 0xF0, 0xF0, 0x00, 0x00, 0xE1, 0xF0, 0xF0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xF0, 
+};
+
+void EnEncount2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnEncount2* this = THIS;
+    s8 pad[4];
+    CollisionHeader* colHeader = NULL;
+
+    BcCheck3_BgActorInit(&this->dynaActor, 0);
+    BgCheck_RelocateMeshHeader(&D_06002420, &colHeader);
+    this->dynaActor.bgId = BgCheck_AddActorMesh(globalCtx, &globalCtx->colCtx.dyna, &this->dynaActor, colHeader);
+    ActorShape_Init(&this->dynaActor.actor.shape, 0.0f, func_800B3FC0, 25.0f);
+    this->dynaActor.actor.colChkInfo.mass = 0xFF;
+    Collider_InitAndSetJntSph(globalCtx, &this->collider, &this->dynaActor, &sJntSphInit, &this->colElement);
+
+    this->dynaActor.actor.targetMode = 6;
+    this->dynaActor.actor.colChkInfo.health = 1;
+    this->scale = 0.1;
+    this->switchFlag = (s16) (this->dynaActor.actor.params & 0x7F);
+
+    if (this->switchFlag == 0x7F) {
+        this->switchFlag = -1;
+    }
+
+    if ((this->switchFlag >= 0) && (Actor_GetSwitchFlag(globalCtx, this->switchFlag) != 0)) {
+        Actor_MarkForDeath(this);
```

Comment:
I think Actor_MarkForDeath wants an actor pointer instead of `this` which is an EnCount2 pointer in this context